tests/baselines/reference/restTupleUnionWithRestContextualParams.errors.txt

```diff
@@ -0,0 +1,43 @@
+tests/cases/compiler/restTupleUnionWithRestContextualParams.ts(7,7): error TS2322: Type '(type: "one" | "two", x: string | number | undefined) => void' is not assignable to type '(...args: [type: "one", x?: number | undefined] | [type: "two", x: string]) => void'.

```

Comment:
This is essentially the same regression that I mentioned [here](https://github.com/microsoft/TypeScript/pull/49218/files#r1071106463). Note that this is only a regression when I take into account the previous version of this PR - this test case doesn't typecheck on the main branch.